lib/active_merchant/billing/gateways/worldpay.rb

```diff
@@ -441,15 +441,23 @@ def add_payment_method(xml, amount, payment_method, options)
       end
 
       def add_network_tokenization_card(xml, payment_method)
+        token_type = {
+          apple_pay: 'APPLEPAY',
+          google_pay: 'GOOGLEPAY',
+          network_token: 'NETWORKTOKEN'
+        }[payment_method.source]
```

Comment:
We usually add something like this as a Constant map, similar to `CARD_CODES` up above.